src/state/debugger.rs

```diff
@@ -10,11 +10,23 @@ use std::collections::{HashMap, HashSet};
 /// Required features:
 /// - `debug`
 pub struct Debugger {
+    /// Single-stepping mode triggers a breakpoint after each instruction
+    single_stepping: bool,
     breakpoints: HashMap<ContractId, HashSet<Word>>,
     last_state: Option<ProgramState>,
 }
 
 impl Debugger {
+    /// Get single-stepping mode
+    pub fn single_stepping(&self) -> bool {
+        self.single_stepping
+    }
+
+    /// Set single-stepping mode
+    pub fn set_single_stepping(&mut self, single_stepping: bool) {
```

Comment:
Maybe file a good first issue to create a debug builder since we are now adding parameters?